### PR TITLE
Add `bounding box` filter for generating (oriented) bounding boxes

### PR DESCRIPTION
**Pull request by user27182**
### Overview

New dataset filter. It's similar to `add_bounding_box` but applies to datasets. The boxes can also be optionally oriented.

### CODE REVIEW

**Comment by user27182** [comment]
Should probably add [`outline`](https://dev.pyvista.org/api/core/_autosummary/pyvista.datasetfilters.outline#pyvista.DataSetFilters.outline) filter to See Also

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Comment by tkoyama010** [comment]
> Should probably add [`outline`](https://dev.pyvista.org/api/core/_autosummary/pyvista.datasetfilters.outline#pyvista.DataSetFilters.outline) filter to See Also

@user27182 Do you want to add this. I don't have a strong opinion about it.

**Comment by user27182** [comment]
> > Should probably add [`outline`](https://dev.pyvista.org/api/core/_autosummary/pyvista.datasetfilters.outline#pyvista.DataSetFilters.outline) filter to See Also
> 
> @user27182 Do you want to add this? I don't have a strong opinion about it.

This was added already in https://github.com/pyvista/pyvista/pull/6524/commits/bd641ed1e86a299d326cdcf0d8b4d1c0fcb128b9 👍

File: tests/core/test_dataset_filters.py [hunk 1]
```diff
@@ -3947,12 +3947,12 @@ def test_bounding_box_return_meta(oriented, as_composite):
     obb, point, axes = box_mesh.bounding_box(
         oriented=oriented, return_meta=True, as_composite=as_composite
     )
-
+    ATOL = 1e-6
```
Comment (tkoyama010) [inline tests/core/test_dataset_filters.py:3950]: Please add a comment as to why you set this value. I think it is a numerical error, but I would like to clarify why it is not the default value.

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: tests/core/test_dataset_filters.py (hunk 1 above)
Comment (user27182) [reply tests/core/test_dataset_filters.py]: Add in https://github.com/pyvista/pyvista/pull/6524/commits/831c555fb4eafaaa958376b038863d9adce95557

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**